### PR TITLE
Favorites pagination

### PR DESCRIPTION
**Pull request by indhu-gunda**
Pagination for the favorites page and change way to determine if favorite in ArtworkCloseUpCard. This was done because now only 1 page of favorites is available in the redux state, so another endpoint was created to find out if a specific artwork was a favorite when it's not available in the page.

### CODE REVIEW

File: backend/src/main/java/com/google/capstone/GetFavoritesCount.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String uid = request.getParameter("uid");
+    Query query = new Query(uid + "Favorites");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    PreparedQuery results = datastore.prepare(query);
+    int count = results.countEntities(FetchOptions.Builder.withDefaults());
+
+    System.out.println(count);
+    System.out.println(uid);
```
Comment (isabelsang) [inline backend/src/main/java/com/google/capstone/GetFavoritesCount.java:32]: remove?

File: backend/src/main/java/com/google/capstone/FindFavorite.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+
+/** Servlet responsible for finding if their is a favorite matching the parameters.
+ *  A Favorite is returned if found, otherwise null.
+ */
```
Comment (katch00) [inline backend/src/main/java/com/google/capstone/FindFavorite.java:25]: Small nit: line 23 should just be /**, 24 should contain "servlet responsible..."

File: backend/src/main/java/com/google/capstone/GetFavoritesCount.java (hunk 1 above)
Comment (indhu-gunda) [reply backend/src/main/java/com/google/capstone/GetFavoritesCount.java]: fixed

File: backend/src/main/java/com/google/capstone/GetFavorites.java [hunk 3]
```diff
@@ -20,16 +20,19 @@
...
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String uid = request.getParameter("uid");
+    int page = getParameter(request, "page", DEFAULT_PAGE);
     Query query = new Query(uid + "Favorites").addSort("timestamp", SortDirection.DESCENDING);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    List<Entity> results = datastore.prepare(query).asList(FetchOptions.Builder.withLimit(MAX_FAVORITES));
+    List<Entity> results = datastore.prepare(query).asList(FetchOptions.Builder
+            .withOffset((page-1) * NUM_FAVORITES)
```
Comment (BearBurg) [inline backend/src/main/java/com/google/capstone/GetFavorites.java:34]: nit: `(page - 1)`

File: backend/src/main/java/com/google/capstone/FindFavorite.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+@WebServlet(name = "findFavorite", value = "/api/v1/find-favorite")
+public class FindFavorite extends HttpServlet {
+
+  private static final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  private static final Gson gson = new Gson();
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
```
Comment (BearBurg) [inline backend/src/main/java/com/google/capstone/FindFavorite.java:34]: Forget the comment about whether we need to find if the artwork has been favorited in the previous PR.

**Pull request merged**